src/frontend/src/components/modal.ts

```diff
@@ -0,0 +1,59 @@
+import { html, TemplateResult, render } from "lit-html";
+
+export const createModal = ({
+  slot,
+  id,
+  confirmText = "Ok",
+}: {
+  slot: TemplateResult;
+  id?: string;
+  confirmText?: string;
+}): HTMLDialogElement => {
+  const modalHtml = html`
+    <form method="dialog" class="c-modal__content c-card c-card--modal">
+      <div class="c-modal__inner">
+        <span class="c-modal__close" data-closemodal>&times;</span>
+        ${slot}
+        <div class="c-modal__footer">
+          <button type="submit" class="c-button c-button--primary">
+            ${confirmText}
+          </button>
+        </div>
+      </div>
+    </form>
+  `;
+
+  const $modal = document.createElement("dialog");
+  $modal.id = id != null ? id : "modal";
```

Comment:
is there a risk to have two or more ids next to each other on the dom? if yes, maybe worth to use a custom attribute instead?

Reply:
the ID is there to be used for accessibility (target)

Reply:
should we prevent the potential error then?

Reply:
we could check for existing ID's but this would happen on runtime.